src/Symfony/Component/Mailer/Command/MailerDebugCommand.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Mailer\Exception\TransportExceptionInterface;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Mime\Email;
+
+/**
+ * Helps making sure your mailer provider is operational.
+ *
+ * @author Guillaume MOREL <me@gmorel.io>
+ *
+ * @final
+ */
+class MailerDebugCommand extends Command
+{
+    /** {@inheritdoc} */
+    protected static $defaultName = 'debug:mailer:send-email';
+
+    /** @var ?MailerInterface */
+    private $mailer;
+
+    /** @var SymfonyStyle */
+    private $io;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct(?MailerInterface $mailer)
+    {
+        parent::__construct();
+
+        $this->mailer = $mailer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isEnabled()
+    {
+        return null !== $this->mailer;
```

Comment:
This allows to pass a regression on [CachePoolsTest](https://github.com/symfony/symfony/blob/87868baacbd7f17197d3948d0482c0c9478ca845/src/Symfony/Bundle/FrameworkBundle/Tests/Functional/CachePoolsTest.php#L20)
But I'm not really proud of having a constructor with nullable args. Not very coherent to have a mandatory dependency optional.

I tried to get some inspiration from [FrameworkExtension](https://github.com/symfony/symfony/blob/cc130e1d9c5af38f89bafed2b576eb05228b0c3e/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L209-L218) without success. But I would rather have the "checks" on the MailerComponent to reduce the coupling with FrameworkBundle.

If you have a better option, I'm taking it.